Create Article: Big-O Notation

angregor opened the pull request:
Part of Issue #1048

angregor committed:
Create Article: Big-O Notation

#1048

alayek commented on Algorithms-Big-O-Notation.md:
```diff
@@ -0,0 +1,54 @@
+# Big-O Notation
+
+In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.
```
> In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.

Great opening, but maybe you would add a line or two explaining what you mean by _limiting behavior_.

alayek commented on Algorithms-Big-O-Notation.md:
```diff
@@ -0,0 +1,54 @@
+# Big-O Notation
+
+In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.
+In short Big-O notation is used to describe the growth or decline of a function, usually with respect to another function.
```
```diff
- In short Big-O notation
+ In short, big-O notation
```

alayek commented on Algorithms-Big-O-Notation.md:
```diff
@@ -0,0 +1,54 @@
+
+In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.
+In short Big-O notation is used to describe the growth or decline of a function, usually with respect to another function.
+For example we say that x = O(x^2) for all x > 1 or in other words, x^2 is an upper bound on x and therefore it grows faster.
```
You might want to toss in a line describing what you mean by the `^` symbol. In this case, it is exponentiation, but in  few common programming languages, its XOR.

angregor committed:
Update Algorithms-Big-O-Notation

Added some updates as suggested by @alayek specifically explaining limiting behavior, fixing some grammatical errors, and adding a note about how exponents are expressed in this article.

angregor committed:
Update Algorithms-Big-O-Notation

More grammatical errors fixed, specifically capitalization errors on 'big-O notation'

atjonathan commented on Algorithms-Big-O-Notation.md:
```diff
@@ -0,0 +1,58 @@
+The symbol of a claim like x = O(x^2) for all x > *n* can be substituted with  x <= x^2 for all x > *n* where *n* is the minimum number that satisfies the claim, in this case 1.
+Effectively, we say that a function f(x) that is O(g(x)) grows slower than g(x) does.
+
+Comparitively, in computer science and software development we can use big-O notation in order to describe the time complexity or efficiency of our algorithm.
```
~~of our algorithm.~~ of algorithms.

atjonathan commented:
### **LGTM :+1:**

angregor committed:
Update Algorithms-Big-O-Notation

Fixed another grammatical error

Rafase282 commented:
# LGTM :+1: 

Hopefully we can expand this later on. I personally would like to learn how to determine performance and all. This is one topic I passed and still learnt nothing in college.

raisedadead commented on Algorithms-Big-O-Notation.md:
```diff
@@ -0,0 +1,58 @@
+It is very apparent that while for small use cases this time complexity might be alright, at a large scale bubble sort is simply not a good solution for sorting.
+This is the power of big-O notation: it allows developers to easily see the potential bottlenecks of their application, and take steps to make these more scalable.
+
+For more information on why big-O notation and algorithm analysis is important visit this [hike](https://www.freecodecamp.com/videos/big-o-notation-what-it-is-and-why-you-should-care)!
```
```diff
-[hike]
+[video challenge]
```

raisedadead commented:
Everything else LGTM.

angregor committed:
Update Algorithms-Big-O-Notation

changed phrasing as per suggestion

atjonathan commented:
@koustuvsinha, anything you wanna add?
